### PR TITLE
Remove dynamic routing and loading

### PR DESCRIPTION
**Pull request by zurgl**
- Remove dynamic routing
- Upgrade to nextjs-12
- Remove dynamic loading
- Make usage of  dynamic loading for **react-json-view** for no regression
- Solve un met dependency warning
- Leverage swc (remove .babelrc) and activate styled on SSR using **next.config.js**

The performance are blazing fast with swc (a rust bundler)

### CODE REVIEW

File: pages/avalanche.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    <>
+      <Head>
+        <title>{`Figment Learn - ${chain.label} Pathway`}</title>
+        <meta name="description" content="Generated by create next app" />
+        <link rel="icon" href="/favicon.ico" />
+      </Head>
+      <WithLayoutAvalanche chain={chain} markdown={markdown} />
+    </>
+  );
+};
```
Comment (guillaumegaluz) [inline pages/avalanche.tsx:29]: same here - we could just pass `<WithLayoutAvalanche chain={chain} markdown={markdown} />` as children of this refactored component

**Review by guillaumegaluz — APPROVED** [review]
i would just reactor all the repeated code and after that 🚢 it

**Comment by guillaumegaluz** [comment]
Just want to make sure you saw my comments. I think you could refactor the repeated code since all the pages basically have 99% of the same code except for CHAINS.X

**Comment by zurgl** [comment]
> Just want to make sure you saw my comments. I think you could refactor the repeated code since all the pages basically have 99% of the same code except for CHAINS.X

**getStaticProps** is not something we can factorize. It's a specific feature of Next.js saying that for this page Next.js needs to statically download some static content to pre-render it at compile time (here md file).
And as we have removed dynamic routing, it's normal to have repeated code it's a tradeoff against performance.

**Pull request merged**